src/relcast.erl

```diff
@@ -1050,11 +1069,111 @@ make_seq(ID, #state{seq_map=SeqMap, ids=A}=State) ->
+            O = case Old of
+                    %% since we're serializing for the first time, we need to make sure that
+                    %% everything gets written out, otherwise we have partial state that
+                    %% won't restore correctly.
+                    BorU when BorU == undefined orelse
+                              is_binary(BorU) ->
+                        lists:map(fun({K, _V}) -> {K, never_ever_match_with_anything} end, S);
+                    _Size when OSize =/= SSize ->
+                        SKeys = maps:keys(StateMap),
+                        OKeys = maps:keys(Old),
+                        Old1 = lists:foldl(fun(L, OM) ->
+                                                   OM#{L => undefined}
+                                           end,
+                                           Old,
+                                           SKeys -- OKeys),
+                        maps:to_list(maps:without(OKeys -- SKeys, Old1));
+                    _ ->
+                        lists:sort(maps:to_list(Old))
+                end,
+            L = lists:zip(S, O),
+            KeyTree =
+                lists:map(
+                  fun({{K, V}, {_, V}}) ->
+                          %% should be a binary
+                          K;
+                     ({{K, V}, {_, OV}}) ->
+                          KeyName = <<Prefix/binary, (atom_to_binary(K, utf8))/binary>>,
+                          case is_map(V) of
+                              true ->
+                                  do_serialize(K, fixup_old_map(OV), V, <<KeyName/binary, "_">>, Transaction);
+                              false ->
+                                  ok = rocksdb:transaction_put(Transaction, KeyName, V),
+                                  K
+                          end
+                  end,
+                  L),
+            [Mod | KeyTree]
+    end.
+
+fixup_old_map(never_ever_match_with_anything) ->
+    undefined;
+fixup_old_map(M) ->
+    M.
+
+do_deserialize(Mod, NewState, Prefix, KeyTree, RocksDB) ->
+    R = fun Rec(Pfix, [_Top | KT], DB) ->
+                lists:foldl(
+                  fun(K, Acc) when is_atom(K) ->
+                          KeyName = <<Pfix/binary, (atom_to_binary(K, utf8))/binary>>,
+                          Term = case rocksdb:get(DB, KeyName, []) of
+                                     {ok, Bin} ->
+                                         Bin;
+                                     not_found ->
+                                         undefined
+                                 end,
+                          Acc#{K => Term};
+                     (L, Acc) when is_list(L) ->
+                          K = hd(L),
+                          KeyName = <<Pfix/binary, (atom_to_binary(K, utf8))/binary, "_">>,
+                          Acc#{K => Rec(KeyName, L, DB)}
+                  end,
+                  #{},
+                  KT);
+            Rec(_, bin, DB) ->
```

Comment:
What is this clause for? the tests don't seem to hit it?

Reply:
it does get hit by the tests, I think that my reloading hackery is breaking coverage.  that's the case where we have a module that is not using the map stuff at all, and serialize returns a binary.

Reply:
Ok